src/components/CityListScreen.js

```diff
@@ -1,25 +1,75 @@
-import React, { Component } from 'react';
 import { FlatList, View, StyleSheet } from 'react-native';
 import { autobind } from 'core-decorators';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import * as cityService from '../services/city';
+import { connect } from 'react-redux';
+import _ from 'lodash';
 import CityListItem from './CityListItem';
+import { getCities } from '../redux/selectors';
+import { fetchCurrentWeatherForCity } from '../redux/actions';
 
 class CityListScreen extends Component {
   static propTypes = {
     navigation: PropTypes.object,
+    cities: PropTypes.array,
+    fetchCurrentWeatherForCityAction: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      refreshing: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchWeatherForCities();
+  }
+
+  @autobind
+  handleRefresh() {
+    this.setState(
+      {
+        refreshing: true,
+      },
+      () => {
+        this.fetchWeatherForCities();
+      }
+    );
+  }
+
+  @autobind
+  fetchWeatherForCities() {
+    const { cities, fetchCurrentWeatherForCityAction } = this.props;
+
+    _.forEach(cities, city => {
+      fetchCurrentWeatherForCityAction(city.value);
+    });
+
+    this.setState({
```

Comment:
this is not gonna work as you expect. Those fetch requests are just fired and will be executed asynchronously. Which means you'll just fire them and then set refreshing state to false. What you want to do is wait for all of them to finish and then set state.

So, you'll either wrap the whole thing in try - catch with await or use 
`Promise.all([requests]).then(() => this.setState({ refreshing: false });`